spectec/spec/8-reduction.watsup

```diff
@@ -387,14 +407,15 @@ rule Step_read/array.fill-oob:
   -- if $(i + n > |$arrayinst(z)[a].FIELD|)
 
 rule Step_read/array.fill-zero:
-  z; (REF.ARRAY_ADDR a) (CONST I32 i) val (CONST I32 0) (ARRAY.FILL x)  ~>  epsilon
+  z; (REF.ARRAY_ADDR a) (CONST I32 i) val (CONST I32 n) (ARRAY.FILL x)  ~>  epsilon
   -- otherwise
+  -- if n = 0
 
 rule Step_read/array.fill-succ:
-  z; (REF.ARRAY_ADDR a) (CONST I32 i) val (CONST I32 $(n + 1)) (ARRAY.FILL x)  ~>
+  z; (REF.ARRAY_ADDR a) (CONST I32 i) val (CONST I32 n) (ARRAY.FILL x)  ~>
     (REF.ARRAY_ADDR a) (CONST I32 i) val (ARRAY.SET x)
-    (REF.ARRAY_ADDR a) (CONST I32 $(i + 1)) val (CONST I32 n) (ARRAY.FILL x)
-
+    (REF.ARRAY_ADDR a) (CONST I32 $(i + 1)) val (CONST I32 $(n-1)) (ARRAY.FILL x)
```

Comment:
Using the side condition n=n'+1 seems to be a good idea, and I'll try rewriting based on it.

Allowing the use of `n+1` on the LHS, on the other hand, would require either the well-defined meaning of `otherwise` that works well with the presence of both `0` and `n+1` on the LHS, or a new writing style that excludes the use of `otherwise`.

Reply:
So I tried introducing new premise `n=n'+1 `, but it ends up duplication of the premises in the case where there are further branching after checking that n = n' + 1, such as `array.copy`, resulting in rather verbose premises.

I'll stick with the version with `$(n-1)` version for now.